packages/grafana-data/src/transformations/fieldReducer.test.ts

```diff
@@ -172,6 +172,58 @@ describe('Stats Calculators', () => {
     }
   });
 
+  it('consistent results for first/last value with NaN', () => {
+    const info = [
+      {
+        data: [NaN, 200, NaN], // first/last value is NaN
+        result: 200,
+      },
+      {
+        data: [NaN, NaN, NaN], // All NaN
+        result: null,
```

Comment:
Not sure if we want special handling in this case (i.e. if all NaN return NaN vs null)